### ISSUE TITLE
Could you please explain how to do prediction using One-Class Deep SVDD?

### ISSUE THREAD

**Issue by HelloSeeing** [lukasruff/Deep-SVDD#7]
Thanks for your great work!
There is a point that I do not understand clearly. Since that the objective function of One-Class Deep SVDD (OC-SVDD) is $$\min \limits_{W} \frac{1}{n} \sum_{i=1}^n ||\phi (x_i; W) - c||^2 + regularization\_term$$, and the anomaly score is $s(x) = ||\phi(x; W^*) - c||^2$, how to know whether an input $x$ is anomaly or not by this score $s(x)$? It seems that there should be a threshold for $s(x)$ to make determination, but how to calculate this threshold? 
Thanks a lot for your reply!

**Comment by holagvk** [comment]
> For soft-boundary Deep SVDD, we can adjust this (s(**x**)) score by subtracting the final radius R∗ of the trained model such that anomalies (points with representations outside the sphere) have positive scores, whereas inliers have negative scores.

Probably this will answer your question.

**Comment by HelloSeeing** [comment]
@gvkalra How to choose the value of R, since that there is no R in the loss function of One-Class Deep SVDD.

**Comment by lukasruff** [comment]
Thanks for your interest and sorry for my late reply.

The score s(**x**) gives you an order of "anomalousness" for each example. The higher the score, the more anomalous the example **x** is w.r.t. to some trained Deep SVDD model.

As @gvkalra pointed out, radius R determines an anomaly threshold for soft-boundary SVDD. In Proposition 4 (ν-property) of the paper, we prove that the optimal radius R is given when a fraction ν of the training data lies outside the sphere. By this, you can include some prior assumption on the fraction of outliers you assume to be in the training data.

For One-Class Deep SVDD, you could similarly determine a threshold by taking the (1-ν)-quantile of the scores you obtain, i.e. again the assumption is that a fraction ν of your data is anomalous. I would recommend to derive this threshold on some hold out validation set.

In addition, by this you also control how conservative your anomaly detector is, which depends on the context of application. If you don't want to miss any anomaly, you would choose a higher value of ν (i.e. a lower quantile).

So in summary, choosing a threshold depends on (1) how polluted (with anomalies) your training data is, and (2) how conservative your anomaly detector should be.

I recommend ν=0.1 (or the 90%-quantile of scores) as a starting point.

**Comment by HelloSeeing** [comment]
Appreciate for your reply. I understand it clearly, and will close this issue.
